codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -401,6 +416,17 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                                 "Multiple content-length headers received");
                     }
                 }
+            } else if (validateHeaders && !headers.isEmpty()) {
```

Comment:
This change broke users of `grpc-netty`.

https://github.com/grpc/grpc-java/pull/10663

This is clearly a `grpc-netty` bug so there's no action required in Netty - I just wanted to comment here to make the maintainers aware of this issue.